contents/blog/github-gitlab-feature-flags.md

```diff
@@ -0,0 +1,94 @@
+---
+title: How GitHub and GitLab use feature flags
+date: 2023-07-26
+author: ["ian-vanagas"]
+showTitle: true
+rootpage: /blog
+sidebar: Blog
+hideAnchor: true
+featuredImage: ../images/blog/green-blog-image.jpg
+featuredImageType: full
+category: Engineering
+tags:
+ - Guides
+ - Feature flags
+---
+
+Feature flags are a simple way to control and rollout features or parts of code. In their simplest form, they enable you to turn on or off features or code paths for certain users. The secret of feature flags is that they have more depth than this, and getting the most out of them at scale requires knowledge and process.
+
+Although we covered feature flag [use cases](/blog/feature-flag-benefits-use-cases) and [best practices](/blog/feature-flag-best-practices), it is always useful to learn how large companies work with feature flags. Two leading examples are GitHub and GitLab. This post details why, when, and how they use feature flags.
+
+## Why do GitHub and GitLab use feature flags?
+
+Both GitHub and GitLab are massive software companies. They are both at the forefront of software development and DevOps, so maintaining best practices is important. Broadly, this means making the deployment of code safer and more consistent.
+
+In GitHub’s case, they frame this as "**lowering deployment risk**." They want deployments to be risk-free. Putting new changes behind a flag lowers the risk of failing deployments. If there is an issue with a feature, you disable the flag rather than needing to roll back the entire deployment.
+
+In GitLab’s case, they have a "**progressive delivery mindset**." They find feature flags are a great tool for incremental, continuous delivery, allowing them to separate deployment and rollout. This is especially valuable to them because deployment is inconsistent due to their self-hostability. Feature flags increase the throughput of changes shipped as well as psychological safety for developers.
```

Comment:
```suggestion
In GitLab’s case, they have a "**progressive delivery mindset**." They find feature flags are a great tool for incremental, continuous delivery, allowing them to separate deployment and rollout. This is especially valuable to them because deployment is inconsistent due to their self-hostability. Feature flags increase the speed of shipping changes as well as provide psychological safety for developers.
```